src/thunderbots/software/util/visualizer_messenger/visualizer_messenger.cpp

```diff
@@ -0,0 +1,162 @@
+/**
+ * Implemenation of visualizer messenger
+ * 
+ * Lasted edited by Muchen He (i@muchen.ca) on 2019-01-26
+ */
+
+#include <iostream>
+
+#include "visualizer_messenger.h"
+
+namespace Util
+{
+
+const LayerMsgMap& VisualizerMessenger::getLayerMap() const
+{
+	return m_draw_layers;
+}
+
+void VisualizerMessenger::clearLayers()
+{
+	// Clears all shape vector in all the layers
+	for (std::pair<std::string, LayerMsg> layer : m_draw_layers)
+	{
+		layer.second.shapes.clear();
+	}
+}
+
+void VisualizerMessenger::ellipse(
```

Comment:
I think `addEllipseToLayer` is redundant because it is already specified in function comment, and implied in the parameters. Many drawing libraries use simple function names: [three.js](https://threejs.org/docs/index.html#api/en/geometries/SphereGeometry), [Processing 3](https://py.processing.org/reference/ellipse.html), [p5.js](https://p5js.org/reference/#/p5/ellipse), [Paper.js](http://paperjs.org/reference/rectangle/), [SFML](https://www.sfml-dev.org/documentation/2.5.1/classsf_1_1Rect.php) to name a few.

Reply:
Probably nit, but in case we need to change the name of the function, perhaps `drawEllipse` would satisfy the verb requirement while staying in line with other libraries (such as Pixi.js)

Reply:
`drawEllipse` would be fine with me. I feel pretty strongly about naming functions as actions either way, it really does make usages of this function _much_ more readable.

Reply:
Cool, I'll change it tonight